src/api/app/models/review.rb

```diff
@@ -19,7 +19,14 @@ class NotFoundError < APIError
   validates :reviewer, length: { maximum: 250 }
   validates :reason, length: { maximum: 65_534 }
 
-  validate :check_initial, on: [:create]
+  validates :user, presence: true, if: :by_user?
+  validates :group, presence: true, if: :by_group?
+  validates :project, presence: true, if: :by_project?
```

Comment:
All this was not enforced before. You would need to make sure that `by_project` is set to nil if the project is removed :thinking:

Reply:
this would break history

Reply:
then this validation is wrong :rofl:

Reply:
I'm afraid bgeuken's idea just doesn't work.